### DISCUSSION TITLE
How to load polyfills based on usage?

### DISCUSSION THREAD

**Discussion by wereHamster** [vercel/next.js#11633] [Help]
I'm trying to load polyfills based on usage, ie. babel preset-env useBuiltIns: usage option. I have a page like this:

```
export default () => {
  console.log(Object.fromEntries([]));
  return null;
};
```

When I compile that JS file with babel cli and the babel config `["@babel/preset-env", { useBuiltIns: "usage", corejs: "3" }]`, then the file that is generated includes all the necessary polyfills. However when I configure next/babel preset-env with the same options, then the polyfills are not loaded.

Essentially: how do I make this not crash in Edge 17: https://codesandbox.io/s/next-polyfill-howto-lqy3x (btw, the useEffect is there because CSB runs an older version of node.js).

**Comment by rafde** [comment]
here's my babel.config.js file
```
module.exports = function( api ) {
	// Cache the returned value forever and don't call this function again.
	api.cache( true );

	return {
		presets: [
			// https://babeljs.io/docs/en/babel-preset-env
			[
				'next/babel',
				{
					'preset-env': {
						corejs: {
							version: 3,
							proposals: true,
						},
						useBuiltIns: 'usage',
					},
				},
			],
		],
	};
};
```
